ExtraDry/Sample.Data/Extensions/TaxonomyExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using System;
+
+namespace Sample.Data {
+    public static class TaxonomyExtensions {
+        /// <summary>
+        /// Moves a subtree from one point in the heirarchy to another.
+        /// </summary>
+        public static async Task MoveSubtree(this ITaxonomyEntity subtreeRootToMove, ITaxonomyEntity newParent, DbContext context)
+        {
+            // Check that the new parent is at the same strata level as the current parent.
+            if(subtreeRootToMove.Strata != newParent.Strata + 1) {
+                throw new ArgumentException("Moving the subtree cannot change it's strata level");
+            }
+
```

Comment:
Are there any other assumptions you need to guard against?  E.g. that the change tracker is not tracking changes to any T except for subtreeRootToMove?

Reply:
I've got a check on the ChangeTracker to make sure there's no pending changes. I don't think they should cause any issues if they were there, but there is a risk, so I've just told it to avoid those changes.